File: programacion/python/aplicacion/clases/consultadbapli.py

import mysql.connector

class consultadbapli:

    def __init__(self, dbconn, perfil):

        self.dbconn = dbconn
        self.perfil = perfil
    
# Consulta de las acciones de un perfil en la base de datos
    def listaaccper(self):
        dbconncursor = self.dbconn.cursor()
        dbconncursor.execute("select accion from desmicaccper where perfil = '%s'" % (self.perfil))
        
        acciones = dbconncursor.fetchall()
        listaacc = []
        for i in range(len(acciones)):
            listaacc.append(acciones[i][0])
        
        return listaacc

File: programacion/php/aplicacion/clases/consultadbapli.php
<?php

namespace clasesapli;

class consultadbapli {

    private $dbconn;
    private $perfil;

    public function __construct($dbconn, $perfil)
    {
        $this->dbconn = $dbconn;
        $this->perfil = $perfil;
    }

// Consulta de las acciones de un perfil en la base de datos
    function listaaccper(){
        $acciones = $this->dbconn->query("select accion from desmicaccper where perfil = '$this->perfil';");
        $listaacc = array();
        while ($row = $acciones->fetch_array()[0]) {
            array_push($listaacc, $row);
        }

        return $listaacc;
    }
}

?>